### PR TITLE
Refactored MysqlEngine, created PdoEngine, added Postgres support

### PR DESCRIPTION
**Pull request by robertscherer**
In reference to #30

### CODE REVIEW

**Comment by robertscherer** [comment]
@josegonzalez The PdoEngine refactoring and Postgres support is done, would be great if you could give it a look.

I'm not sure what to do about the coveralls error - because effectively, test coverage didn't decrease.

File: tests/josegonzalez/Queuesadilla/Engine/AbstractPdoEngineTest.php [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+        $Engine = new $this->engineClass($this->Logger, $this->url);
+        $this->assertNotNull($Engine->connection());
+
+        $Engine = new $this->engineClass($this->Logger, $this->config);
+        $this->assertNotNull($Engine->connection());
+    }
+
+    /**
+     * @covers josegonzalez\Queuesadilla\Engine\PdoEngine::connect
```
Comment (josegonzalez) [inline tests/josegonzalez/Queuesadilla/Engine/AbstractPdoEngineTest.php]: I think the reason test coverage went down is that the MysqlEngine is no longer covered here.

**Comment by robertscherer** [comment]
Here we go :)

File: src/josegonzalez/Queuesadilla/Engine/PdoEngine.php [hunk 2]
```diff
@@ -229,22 +229,17 @@ public function quoteIdentifier($identifier)
             return $this->startQuote . $identifier . $this->endQuote;
         }
 
+        // string.string
```
Comment (josegonzalez) [inline src/josegonzalez/Queuesadilla/Engine/PdoEngine.php]: Why not instead pull the regex out into a variable? That should help explain what the regex is for.

**Comment by josegonzalez** [comment]
Thanks!

**Pull request merged**